rust/agama-dbus-server/src/network/nm/client.rs

```diff
@@ -122,6 +126,24 @@ impl<'a> NetworkManagerClient<'a> {
         Ok(())
     }
 
+    /// Deactivates a NetworkManager connection.
+    ///
+    /// * `path`: D-Bus patch of the connection.
+    async fn deactivate_connection(&self, path: OwnedObjectPath) -> Result<(), ServiceError> {
+        let proxy = NetworkManagerProxy::new(&self.connection).await?;
+        match proxy.deactivate_connection(&path.as_ref()).await {
+            Err(e) => {
+                // Ignore ConnectionNotActive error since this just means the state is already correct
+                if e.to_string().contains("ConnectionNotActive") {
+                    Ok(())
+                } else {
+                    Err(ServiceError::DBus(e))
+                }
+            }
+            _ => Ok(()),
+        }
```

Comment:
I am unsure about this part. This is needed for the migration otherwise it would show the `ConnectionNotActive` error when creating the connection with the state initially set to down. This was the best way I found but maybe there is a better way?

Reply:
I do not think we get a specific error from the D-Bus service, so your approach looks good.